client/me/account/email-address.jsx

```diff
@@ -0,0 +1,268 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import emailValidator from 'email-validator';
+import { localize } from 'i18n-calypso';
+import { noop } from 'lodash-es';
+
+/**
+ * Internal dependencies
+ */
+import { cancelPendingEmailChange, setUserSetting } from 'state/user-settings/actions';
+import FormTextInput from 'components/forms/form-text-input';
+import FormTextValidation from 'components/forms/form-input-validation';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import getOriginalUserSetting from 'state/selectors/get-original-user-setting';
+import getUnsavedUserSettings from 'state/selectors/get-unsaved-user-settings';
+import getUserSetting from 'state/selectors/get-user-setting';
+import isPendingEmailChange from 'state/selectors/is-pending-email-change';
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+class AccountSettingsEmailAddress extends React.Component {
+	static propTypes = {
+		debug: PropTypes.func,
+		emailValidationListener: PropTypes.func,
+		errorNotice: PropTypes.func.isRequired,
+		focusHandler: PropTypes.func,
+		isSubmitting: PropTypes.func.isRequired,
+		successNotice: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+		userSettings: PropTypes.object,
+	};
+
+	static defaultProps = {
+		debug: noop,
+		focusHandler: noop,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			emailValidationError: false,
+			initialHasPendingEmailChange: this.hasPendingEmailChange( props ),
+			isCancellingPendingEmailChange: false,
+		};
+		this.cancelEmailChange = this.cancelEmailChange.bind( this );
+		this.updateEmailAddress = this.updateEmailAddress.bind( this );
+		// Flag the address as invalid if it is unchanged.
+		props.emailValidationListener &&
+			props.emailValidationListener( this.isEmailAddressUnchanged( props ) );
+	}
+
+	getEmailAddress( props ) {
+		if ( this.hasPendingEmailChange( props ) ) {
+			return this.getPendingEmailAddress( props );
+		}
+		if ( props.userSettings ) {
+			return props.userSettings.getSetting( 'user_email' );
+		}
+		if ( props.unsavedUserSettings && typeof props.unsavedUserSettings.user_email === 'string' ) {
```

Comment:
I think there might be still something wrong with this logic (or the way we handle the data in Redux/flux?). There's a different behaviour still for the empty email case.

In `/me/account`, when the field is empty, there's an explicit error message shown:
<img width="749" alt="Screenshot 2020-07-14 at 14 16 36" src="https://user-images.githubusercontent.com/3392497/87436781-05902b00-c5dd-11ea-8e30-6f196f084e85.png">

However, in `account-email`, there is no error message:
<img width="750" alt="Screenshot 2020-07-14 at 14 15 01" src="https://user-images.githubusercontent.com/3392497/87436836-18a2fb00-c5dd-11ea-92a8-3ec307848e45.png">

The field is highlighted as error, though - and you can't submit the form... unless you go to a different view and come back:
<img width="756" alt="Screenshot 2020-07-14 at 14 17 50" src="https://user-images.githubusercontent.com/3392497/87436873-22c4f980-c5dd-11ea-9714-cc2610e5f771.png">

😁

Reply:
This was basically the same underlying logic issue around `''` being falsey that I worked on in 6e6d760, but I didn't catch the problem in the validation checks. Fixed now.